### PR TITLE
feat(reports) include traffic entry types

### PR DESCRIPTION
**Pull request by kikito**
Previously the anonymous reports ping only included a single counter for the requests the server did:

```
requests=100
```

With this PR, it also differentiates between types:
```
requests=100;http_reqs=10;https_reqs=0;grpc_reqs=20;grpcs_reqs=0;ws_reqs=70;wss_reqs=0
```

### CODE REVIEW

File: kong/reports.lua [hunk 1]
```diff
@@ -150,6 +156,83 @@ local function create_timer(...)
 end
 
 
+local function get_counter(key)
+  local count, err = kong_dict:get(key)
+  if err then
+    log(WARN, "could not get " .. key .. " from 'kong' shm: ", err)
```
Comment (thibaultcha) [inline kong/reports.lua]: Please use the variadic arguments form instead of Lua-land concatenations

File: kong/reports.lua [hunk 2]
```diff
@@ -150,6 +156,83 @@ local function create_timer(...)
...
+  end
+  return count or 0
+end
+
+
+local function reset_counter(key, amount)
+  local ok, err = kong_dict:incr(key, -amount, amount)
```
Comment (thibaultcha) [inline kong/reports.lua]: Better use `:safe_set(key, 0)` to avoid race-conditions in between the previous `get()` and this `incr()` step.

Comment (kikito) [reply kong/reports.lua]: It puzzled me a bit too, but I think we *want* to preserve these "race conditions".

This is my understanding of what's happening now:

```
                Flow of Time
                    ||| 
                    VVV 

      Worker A       |     Worker B
                     |
                     |
get_counter -> 100   |
                     |  <log phase> incr_counter(1) -> 101
                     |
reset_counter(-100)  |
============================================
                     
Final counter value: 1 (correct, the incr_counter from Worker B was preserved)
```

Now with what you propose:
```
                Flow of Time
                    ||| 
                    VVV 

      Worker A       |     Worker B
                     |
                     |
get_counter -> 100   |
                     |  <log phase> incr_counter(1) -> 101
                     |
safe_set(0)          |
============================================
                     
Final counter value: 0 (incorrect, the incr_counter from Worker B was lost)
```

**Comment by kikito** [comment]
@james-callahan I believe that's not possible currently: you would have to be able to increase a shared dict in stream mode and then read it in http mode.

File: kong/reports.lua (hunk 2 above)
Comment (thibaultcha) [reply kong/reports.lua]: @kikito You are correct! My bad. I should know, this was my implementation -_- https://github.com/Kong/kong/commit/2ace69179fba2b3c193bd3cfb73f4102514ab306#diff-10418ffd13cdf2bc868b10cce41a8781R169

Comment (kikito) [reply kong/reports.lua]: I have added the explanation as a comment in the code. Hopefully it will help whomever visits this code in another 3 years :)

**Pull request merged**